### ISSUE TITLE
GM reports wrong size of array when modyfing it inside "catch"

### ISSUE THREAD

**Issue by gnysek** [YoYoGames/GameMaker-Bugs#7776]
### Description
I've found an issue, that when doing try/catch and changing 2D array value, GM would throw `Variable Index [3] out of range [3]` error, like it would not see elements from place where exception happened.

What's more interesting, it works when ran in debug mode.

Short example:
```gml
try {
a[i][0] = sth;
} catch {
a[i][0] = false;  // it would throw "Variable Index [...] out of range [...]", where ... would be equal "i"
}
```

### Steps To Reproduce
1. Run this script:
```gml
globalvar CHALL;

enum chall {
	unlocked
}

CHALL[0][chall.unlocked] = undefined;
CHALL[1][chall.unlocked] = undefined;
CHALL[2][chall.unlocked] = undefined;
CHALL[3][chall.unlocked] = undefined;
CHALL[4][chall.unlocked] = undefined;
CHALL[5][chall.unlocked] = undefined;

data_from_save = [true, true, true];

show_debug_message("array CHALL before");
show_debug_message(CHALL);

for (var i = 0; i < array_length(CHALL); ++i) {
	try {
		CHALL[i][chall.unlocked] = data_from_save[i];
	}
	catch (e) {
		show_debug_message("-- EXCEPTION! --");
		show_debug_message(CHALL);
		CHALL[i][chall.unlocked] = false; // here it breaks
	}
}

show_debug_message("array CHALL after");
show_debug_message(CHALL);
```
See error (as there's no `data_from_save[3]`).

2. Run it again in debug mode. It would work now.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.8.0.169 Runtime v2024.8.0.216

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
Windows

[b0db078f-d64d-4971-ad09-2b3427d8abfe](https://api.gamemaker.io/api/github/downloads/b0db078f-d64d-4971-ad09-2b3427d8abfe)

**Comment by rwkay** [comment]
Fixed in 2024.11

**Comment by Emc1923** [comment]
Verified in  IDE v2024.1100.0.639  Runtime v2024.1100.0.663.
